src/video/emscripten/SDL_emscriptenvideo.c

```diff
@@ -293,12 +301,31 @@ static bool Emscripten_CreateWindow(SDL_VideoDevice *_this, SDL_Window *window,
     }
 
     selector = SDL_GetHint(SDL_HINT_EMSCRIPTEN_CANVAS_SELECTOR);
-    if (!selector) {
-        selector = "#canvas";
+    if(!selector) {
+        selector = SDL_GetStringProperty(props, SDL_PROP_WINDOW_CREATE_EMSCRIPTEN_CANVAS_ID, "#canvas");
     }
-
     wdata->canvas_id = SDL_strdup(selector);
 
+    selector = SDL_GetHint(SDL_HINT_EMSCRIPTEN_KEYBOARD_ELEMENT);
+    if(!selector) {
+        selector = SDL_GetStringProperty(props, SDL_PROP_WINDOW_CREATE_EMSCRIPTEN_KEYBOARD_ELEMENT, EMSCRIPTEN_EVENT_TARGET_WINDOW);
+    }
+
+    // Emscripten's event targets are not defined as actual strings
+    // For Clarification:
+    // #define EMSCRIPTEN_EVENT_TARGET_INVALID        0
+    // #define EMSCRIPTEN_EVENT_TARGET_DOCUMENT       ((const char*)1)
+    // #define EMSCRIPTEN_EVENT_TARGET_WINDOW         ((const char*)2)
+    // #define EMSCRIPTEN_EVENT_TARGET_SCREEN         ((const char*)3)
+    if(Emscripten_IsEventTarget(selector)) {
+        wdata->keyboard_target = selector;
+        wdata->keyboard_element_is_string = false;
+    }
+    else {
+        wdata->keyboard_element = SDL_strdup(selector);
```

Comment:
Where is this freed?

Reply:
Added free call to `Emscripten_DestroyWindow`